File: PetShop/PetShop.Model/Novost.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace PetShop.Model
{
    public class Novost
    {
        public int Id { get; set; }
        public string Naslov { get; set; }
        public string Tekst { get; set; }
        public DateTime Datum { get; set; }
        public byte[] Slika { get; set; }
     
        public int KorisnikId { get; set; }

        public virtual Korisnik Korisnik { get; set; }
    }
}

File: petshop_mobile/lib/models/Novost.dart
import 'dart:convert';

class Novost {
  final int id;
  final String? naslov;
  final String? tekst;
  final DateTime? datum;
  final List<int>? slika;
  final int? korisnikId;

  Novost(
      {required this.id,
        required this.naslov,
        required this.tekst,
        required this.datum,
        required this.slika,
        required this.korisnikId
      });

  factory Novost.fromJson(Map<String, dynamic> json) {
    return Novost(
      id: int.parse(json["id"].toString()),
      naslov: json["naslov"],
      tekst: json["tekst"],
      korisnikId: int.parse(json["korisnikId"].toString()),
      datum: DateTime.tryParse(json['datum']),
      slika: json["slika"] != null
            ? base64.decode(json['slika'] as String)
            : null
    );
  }

  Map<String, dynamic> toJson() => {
    "id": id,
    "naslov": naslov,
    "tekst": tekst,
    "korisnikId": korisnikId,
    "datum":
    datum == null ? null : datum!.toIso8601String(),
    "slika": slika,
  };
}
